docs/how-to/how-to-name-secret.md

```diff
@@ -0,0 +1,14 @@
+# How to name a secret
+
+This tutorial describes how to name a secret in the Google Secret Manager.
+
+## Naming convention
+
+1. The secret name should be in the following format: `<tool for which secret was created>-<entity>`. For example: `github-kyma-bot-token`. From the name, it should be clear what the secret contains: a GitHub token for the Kyma bot service account.
+   The name should not contain information about where the secret is used. For example: `security-dashboard-github-kyma-bot-token` contains information about the Security Dashboard, which is not necessary. The same secret can be used in different places, so the name should be generic.
+   The secret should contain the following fields:
+    - **secret value** - the value of the secret, for example, the GitHub token.
+    - **description** - the description of the secret in the **Annotations description** field. For example: `GitHub token for the Kyma bot service account`.
+
+2. Apply the `owner` label to the secret in Secret Manager to help identify the secret owner. For example: `owner: neighbors`.
+3. Apply the `type` label to the secret in Secret Manager to help identify the secret type. For example: `type: service-account-token`.
```

Comment:
Can we have `tool`, `component` and `entity` labels? This can be used to filter all secrets for different parts of our systems.

Reply:
What if the secret will be used for multiple tools and components?

Reply:
This labels should not describe a locations or usages of the secret. They purpose is to give us a easy way to filter secrets for the same components. I expect the value should be the same as you have in the name.

Reply:
so in the example from pull request what should be tool, component and entity values?

for me tool is Github, component is empty and entity is kyma-bot-token.

Reply:
do you have any real example where we need to specify tool and component for the secret?

Reply:
Which example you mean? `github-kyma-bot-token`? In my opinion the usage of kyma-bot-token for multiple tools and applications interacting with github.com is wrong. That's why we started creating additional tokens.

Using the real world example.

We use this kyma-bot-token in prow. So the secret name could be a `prow-kyma-bot-github-token` assuming our prow doesn't have different components composing it or we need to use one identity for whole prow application.

As example of need for tool and component real world example we can use an `oci-image-builder`. It needs to talk to GCP so needs a service account which represents it's identity in GCP. I would name this SA like `image-builder-oci-image-builder-gcp-sa-key`. The tool is `image-builder`, the component is `oci-image-builder`, the entity is `gcp-sa`.

If we need the oci-image-builder to integrate with github.com and we would have a secret containing a token for it, I would name it something like `image-builder-oci-image-builder-oci-builder-bot-github-token`.

Reply:
To separate a parts of name we could use an underscore to separate them and say that all underscores in used names must be replaced with dashes. Example: `image-builder_oci-image-builder_gcp-sa_key`

Reply:
So description of https://github.com/kyma-project/test-infra/issues/6536 issue is wrong. Your examples use place where secret is used and issue says it should not contain this information. I think we can have secrets which will be used in the multiple tools and components. Let's refine the issue again in the team.

Reply:
Issue description is correct. I think you are mixing scenarios where you use a name of a tool as an identity and as a description where it is used.

Let's take an existing example. We have a secret with name `workloads_default_kyma-bot-github-sap-token`. It contains a kyma-bot-github-sap-token which is used in workloads clusters in default namespace.
But there is a secret with the same token but used in trusted cluster default namespace.
<img width="300" alt="image" src="https://github.com/kyma-project/test-infra/assets/12400578/2ecb4c22-d462-4356-b2dc-6fdba69dbd8d">

Another example is with sap-slack-bot-token
<img width="265" alt="image" src="https://github.com/kyma-project/test-infra/assets/12400578/8a3092ed-2c4a-4be6-98d9-1a16c015b4b2">


The issue says we should not do that, because we have two secrets containing the same secret.
For sap-slack-bot-token it would be enough to name the secret `prow-slack-bot-token` assuming we use this bot token only as an identity of prow in slack.

Reply:
thank you for explanation, please review again